sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,654 @@
+            n_features is the number of features.
+
+        **predict_params : dict of str -> obj
+            Parameters to the `predict` called by the `final_estimator`. Note
+            that this may be used to return uncertainties from some estimators
+            with `return_std` or `return_cov`. Be aware that it will only
+            accounts for uncertainty in the final estimator.
+
+        Returns
+        -------
+        y_pred : ndarray of shape (n_samples,) or (n_samples, n_output)
+            Predicted targets.
+        """
+
+        check_is_fitted(self)
+        return self.final_estimator_.predict(
+            self.transform(X), **predict_params
+        )
+
+
+class StackingClassifier(_BaseStacking, ClassifierMixin):
+    """Stacked of estimators using a final classifier.
+
+    Stacked generalization consists in stacking the output of individual
+    estimator and use a classifier to compute the final prediction. Stacking
+    allows to use the strength of each individual estimator by using their
+    output as input of a final estimator.
+
+    Note that `estimators` are fitted on the full `X` while `final_estimator`
+    is trained using cross-validated predictions of the base estimators using
+    `cross_val_predict`.
+
+    When `predict_proba` is used by each estimator (i.e. most of the time for
+    `stack_methodd='auto'` or specifically for `stack_methodd='predict_proba'`)
+    and if the classes are a binary (e.g. 0 and 1), one of the column predicted
+    by each estimator will be dropped since they are collinear.
+
+    .. versionadded:: 0.22
+
+    Read more in the :ref:`User Guide <stacking>`.
+
+    Parameters
+    ----------
+    estimators : list of (str, estimator)
+        Base estimators which will be stacked together. Each element of the
+        list is defined as a tuple of string (i.e. name) and an estimator
+        instance. An estimator can be set to None or 'drop' using `set_params`.
+
+    final_estimator : estimator, default=None
+        A classifier which will be used to combine the base estimators.
+        The default classifier is a `LogisticRegression`.
+
+    cv : int, cross-validation generator or an iterable, default=None
+        Determines the cross-validation splitting strategy. Possible inputs for
+        cv are:
+
+        * None, to use the default 5-fold cross validation,
+        * integer, to specify the number of folds in a (Stratified) KFold,
+        * An object to be used as a cross-validation generator,
+        * An iterable yielding train, test splits.
+
+        For integer/None inputs, if the estimator is a classifier and y is
+        either binary or multiclass, `StratifiedKFold` is used. In all other
+        cases, `KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    stack_method : {'auto', 'predict_proba', 'decision_function', 'predict'}, \
```

Comment:
I like this interface much better, not sure what the others think.

Reply:
Much better than what alternative?